src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -73,28 +76,72 @@ private boolean isReady(BlobItem blob, String containerName) {
     }
 
     private void processBlob(BlobClient blob) {
-        envelopeService
-            .findLastEnvelope(blob.getBlobName(), blob.getContainerName())
+        getLastEnvelope(blob)
             .ifPresentOrElse(
                 envelope -> {
                     if (envelope.status == Status.CREATED) {
-                        leaseAndThen(blob, () -> blobProcessor.continueProcessing(envelope.id, blob));
+                        leaseAndThen(blob, () ->
+                            continueProcessingEnvelopeIfItStillExists(
```

Comment:
How about `continueProcessingEnvelopeIfEligible` or sth like this? Based on the code, existing alone is not sufficient for an envelope to be processed.